etl/steps/data/garden/demography/2022-11-30/life_expectancy.py

```diff
@@ -0,0 +1,321 @@
+COLUMNS_IDX = ["country", "year"]
+# age groups considered besides at 0 (at birth)
+AGES_EXTRA = ["15", "65", "80"]
+# year when UN WPP data starts
+YEAR_WPP_START = 1950
+# year separating historical and projection data in UN WPP dataset.
+YEAR_HIST_PROJ = 2022
+# Versioning
+VERSION = "2022-11-30"
+# Cold start
+# The first time this is executed, no metadata file is available. It is created on the fly, during execution time.
+# Once this is done, we create the metadata YAML file using etl-metadata-export command.
+# From then, we use the metadata in that YAML file, which might have some manual edits.
+COLD_START = False
+
+
+def run(dest_dir: str) -> None:
+    log.info("life_expectancy.start")
+
+    # read datasets from garden
+    ds_wpp = Dataset(GARDEN_WPP_DATASET)
+    ds_hmd = Dataset(GARDEN_HMD_DATASET)
+    ds_zij = Dataset(GARDEN_ZIJDEMAN_DATASET)
+    ds_ril = Dataset(GARDEN_RILEY_DATASET)
+    # group datasets into single list
+    all_ds = [ds_wpp, ds_hmd, ds_zij, ds_ril]
+
+    # create tables (all-years, historical and projections)
+    log.info("life_expectancy: create table with all-years data")
+    tb = make_table(ds_wpp, ds_hmd, ds_zij, ds_ril)
+    log.info("life_expectancy: create table with historical data")
+    tb_historical = make_table(ds_wpp, ds_hmd, ds_zij, ds_ril, only_historical=True)
+    log.info("life_expectancy: create table with projection data")
+    tb_projection = make_table(ds_wpp, ds_hmd, ds_zij, ds_ril, only_projections=True)
+
+    # create dataset
+    ds_garden = Dataset.create_empty(dest_dir)
+    if COLD_START:
+        ds_garden.metadata = make_metadata(all_ds)
+    else:
+        ds_garden.metadata.update_from_yaml(N.metadata_path)
+
+    # add tables to dataset
+    log.info("life_expectancy: add tables to dataset")
+    ds_garden.add(tb)
+    ds_garden.save()
+    ds_garden.add(tb_historical)
+    ds_garden.save()
+    ds_garden.add(tb_projection)
+    ds_garden.save()
+
+    log.info("life_expectancy.end")
+
+
+def make_table(
+    ds_wpp: Dataset,
+    ds_hmd: Dataset,
+    ds_zij: Dataset,
+    ds_ril: Dataset,
+    only_historical: bool = False,
+    only_projections: bool = False,
+) -> Table:
+    """Create table.
+
+    Joins all different sources into a single dataframe.
+
+    By default, it creates a table with all years. Use `only_historical` and `only_projections` to create tables with
+    only historical data or only projections, respectively.
+    """
+    log.info("life_expectancy.make_table")
+    # Build DataFrames
+    df_wpp = load_wpp(ds_wpp)
+    df_hmd = load_hmd(ds_hmd)
+    df_zij = load_zijdeman(ds_zij)
+    df_ril = load_riley(ds_ril)
+    df = merge_dfs(df_wpp, df_hmd, df_zij, df_ril)
+
+    # Filter
+    assert not (only_historical and only_projections), "Both only_historical and only_projections can't be True!"
+    if only_historical:
+        df = df[df.index.get_level_values("year") <= YEAR_HIST_PROJ]
```

Comment:
I suppose the current year should be included as a projection, since it's never going to be completed on a given update.